src/puppeteer_utils.js

```diff
@@ -394,6 +394,8 @@ export const gotoExtended = async (page, request, gotoOptions = {}) => {
  *   If true, it will scroll up a bit after each scroll down. This is required on some websites for the scroll to work.
  * @param {string} [options.buttonSelector]
  *   Optionally checks and clicks a button if it appears while scrolling. This is required on some websites for the scroll to work.
+ * @param {function} [options.stopScrollCallback]
+ *   Expected to be an async function. If this function returns `true`, quit scrolling loop.
```

Comment:
1. I would instead say that it "supports" async functions. If you await sync function, it normally works so it can be used too.
2. I would say that the function is called after every scroll.
3. I guess there are no interesting parameters that we should pass in so that should be fine as it is. (the caller should have access to everything already)